### PR TITLE
Made documentation more modular

### PR DESCRIPTION
**Pull request by Snipey**
## Description

Added sections for examples and made the documentation more modular
Added information to homepage of documentation

## Context

Needed to make documentation more readable and ease of adding more examples outside of a massive readme file.

### CODE REVIEW

File: docs/readme.md [hunk 1]
```diff
@@ -22,3 +22,47 @@ footer: MIT Licensed | Copyright © 2018 - Jamiepine
...
+- ⚙️ Modular structure using "collections"
+- ⚡ Cached data & filters with dependency based regeneration
+- ✨ Automatic data normalization
+- 🔒 Model based data validation
+- 🕰️ History tracking with smart undo functions
+- 🔮 Create data relations between collections
+- 🤓 Database style functions
+- 💎 SSOT architecture (single source of truth)
+- 📕 Error logging & snapshot bug reporting
+- 🔧 Wappers for helpers, utilities and service workers
```
Comment (TimothyCole) [inline docs/readme.md]: Can this actually get changed to `Wrappers` please.
Based on: #27

File: docs/getting-started/setup.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+---
+
+## Vanilla Setup
+
+Manually setting up pulse without a framework
+
+```js
+import Pulse from 'pulse-framework';
+
+new Pulse.Library({
```
Comment (timokoenig) [inline docs/getting-started/setup.md]: Vanilla Setup won't work without a basic storage implementation. Maybe we should add something like this

```
storage: {
    async: true,
    store: {},
    get: function(key) {
      return this.store[key];
    },
    set: function(key, value) {
      this.store[key] = value.toString();
    },
    clear: function() {
      this.store = {};
    },
    remove: function(key) {
      delete this.store[key];
    }
  },
```

Comment (TimothyCole) [reply docs/getting-started/setup.md]: By default pulse uses the localStorage system for persisting data.
No storage object needs to be defined for it to work.
Example: https://codesandbox.io/s/pulse-vanilla-8x8og?expanddevtools=1&fontsize=14&module=%2Fsrc%2Findex.js

Comment (timokoenig) [reply docs/getting-started/setup.md]: You're right as long as you're web based. React native doesn't have the window.localStorage

Comment (TimothyCole) [reply docs/getting-started/setup.md]: Very true. Though this example is for vanilla not React or ReactNative.
We'd greatly appreciate additions to the docs if you'd like to start the ReactNative section.

Comment (timokoenig) [reply docs/getting-started/setup.md]: Sounds good. I will write something together. Additions coming soon 👍

Comment (TimothyCole) [reply docs/getting-started/setup.md]: Awesome; Thank you so much for the help! :smiley:

File: docs/readme.md (hunk 1 above)
Comment (Snipey) [reply docs/readme.md]: Fixed typo

File: docs/readme.md [hunk 3]
```diff
@@ -21,4 +21,40 @@ footer: MIT Licensed | Copyright © 2018 - Jamiepine
...
+**React & React Native support coming soon!**
 :::
+
+# Pulse
+
+Pulse is an application logic library for reactive Javascript frameworks with support for VueJS, React and React Native. Lightweight, modular and powerful, but most importantly easy to understand.
+
+## Why Pulse?
+
+After exploring the many options for Javascript state libraries, including the popular VueX and Redux, I felt like I needed a simpler solution. I wanted to get more out of a library than just state management― something that could provide solid structure for the **entire** application. It needed to be stuctured and simple, but also scalable. This library provides everything needed to get a reactive javascript front-end application working fast, taking care to follow best practices and to employ simple terminology that makes sense even to beginners.
```
Comment (Jackdouglas99) [inline docs/readme.md]: > It needed to be stuctured and simple
the correct spelling is "stuctured"

**Comment by Snipey** [comment]
Should this be sitting on https://jamiepine.github.io/pulse/ or https://pulsejs.org ????

**Pull request merged**